### PR TITLE
NCN-4: /api/articles/:article_id - Get article by ID, custom error, psql error.

### PR DESCRIPTION
**Pull request by aclearskye**

### CODE REVIEW

File: __tests__/api-article_id.test.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+          votes: 100,
+        };
+        expect(body).toEqual(expected);
+        expect(body.hasOwnProperty("title")).toBe(true);
+        expect(body.hasOwnProperty("topic")).toBe(true);
+        expect(body.hasOwnProperty("author")).toBe(true);
+        expect(body.hasOwnProperty("body")).toBe(true);
+        expect(body.hasOwnProperty("created_at")).toBe(true);
+        expect(body.hasOwnProperty("votes")).toBe(true);
+        expect(body.hasOwnProperty("eggyBread")).toBe(false);
```
Comment (alexdevdavis) [inline __tests__/api-article_id.test.js]: worth considering to what extent these are redundant lines of code, given the preceding expect equality

File: __tests__/api-article_id.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      });
+  });
+  it("should respond with a 400 error when given an ID that is not a number", () => {
+    return request(app)
+      .get("/api/articles/eggsForBreakfast")
+      .expect(400)
+      .then(({ body }) => {
+        expect(body).toEqual({ msg: "Bad Request!" });
+        expect(body.hasOwnProperty("msg")).toBe(true);
+        expect(body.hasOwnProperty("eggyBread")).toBe(false);
```
Comment (alexdevdavis) [inline __tests__/api-article_id.test.js]: eggyBread seems to be sneaking in all around this place. Consider whether eggyBread, though delicious, is actually contributing meaningfully to your test ... don't want to end up with egg on one's face

File: middleware/errors.js [hunk 3]
```diff
@@ -1,4 +1,11 @@
 exports.errorMiddleware = (err, req, res, next) => {
-  console.log(err);
+  if (err.status && err.msg) {
+    res.status(err.status).send({ msg: err.msg });
+    next(err);
+  } else if (err.code === "22P02") {
+    res.status(400).send({ msg: "Bad Request!" });
+  } else {
+    next(err);
```
Comment (alexdevdavis) [inline middleware/errors.js]: think about what this code is actually doing, and how it might introduce unintended behaviour

**Review by alexdevdavis — CHANGES_REQUESTED** [review]
Nicely done :) Have a look at what your conditionals are doing in errors.js, refactor, then merge

File: model/article_id-model.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const db = require("../db/connection");
+
+function validateID(id) {
+  if (isNaN(parseInt(id))) {
+    throw { status: 400, msg: "Bad Request!" };
+  }
+}
```
Comment (scar1377) [inline model/article_id-model.js:7]: not an id can be handled by PSQL Error Handler

**Review by scar1377 — APPROVED** [review]
Looks good! The 400 error can be handled by PSQL error handler. You can update it if you like. Or can just go ahead and merge to main

File: model/article_id-model.js (hunk 4 above)
Comment (aclearskye) [reply model/article_id-model.js:7]: i did use a psql error handler but i like the way the error logic is being dealt in the model and it gets passed to my error handler in my errors.js file ill merge now and for my next tix ill incorporate some PSQL error handling!! :D

**Pull request merged**